ndcube/mixins/plotting.py

```diff
@@ -406,21 +409,32 @@ def _animate_cube_1D(self, plot_axis_index=-1, axes_coordinates=None,
                           ylabel="Data [{0}]".format(data_unit), **kwargs)
         return ax
 
-    def _derive_axes_coordinates(self, axes_coordinates, axes_units):
+    def _derive_axes_coordinates(self, axes_coordinates, axes_units, data_shape, edges=False):
         new_axes_coordinates = []
         new_axes_units = []
         default_labels = []
         default_label_text = ""
         for i, axis_coordinate in enumerate(axes_coordinates):
             # If axis coordinate is None, derive axis values from WCS.
             if axis_coordinate is None:
-                # N.B. This assumes axes are independent.  Fix this before merging!!!
-                new_axis_coordinate = self.axis_world_coords(i)
-                axis_label_text = self.world_axis_physical_types[i]
+                # Fix: We would downscale the dependent data into the shape of the axes.
+                xdata = self.axis_world_coords(i, edges=edges)
```

Comment:
Since this function is not only used to find the x-axis, this variable should probably not be name ```xdata```.

Reply:
Naming `data` should work?

Reply:
It's not data though either, in the sense it's not from the data array.  It's coordinate info.  Since the for loop is enumerating through each ```axis_coordinate```, call this variable ```new_axis_coordinate```.  I see you call it that in the end anyway.

Reply:
Shall I rename it to `new_axis_coordinate`? That seems intuitive.